### DISCUSSION TITLE
unban failure

### DISCUSSION THREAD

**Discussion by kuhytitusdlz** [umputun/tg-spam#67] [Q&A]
Hi umputun,

I used docker-compose to deploy the application, and when I try to unban a user I get a warning

![image](https://github.com/umputun/tg-spam/assets/15821799/428f2a5d-8c12-40bc-a48e-f65d75c0a591)

tg-spam    | 2024/03/24 11:10:32.014 [WARN]  failed to process callback: failed to unban user: failed to update ham for "Нужен человек для удаленного заработка. С вас, телефон и 2 часа свободного времени. Доход достойный. Для связи (+) в ЛС": can't update ham samples: can't update ham samples: failed to open /srv/data/ham-dynamic.txt: open /srv/data/ham-dynamic.txt: no such file or directory

**Comment by kuhytitusdlz** [comment]
*I don't see this file /srv/data/ham-dynamic.txt in the container

```bash
/srv $ ls -la ./data
total 244
drwxrwxr-x    2 app      app           4096 Mar 24 11:09 .
drwxr-xr-x    1 root     root          4096 Mar 24 11:07 ..
-rwxrwxr-x    1 app      app             58 Mar 24 07:20 .not_mounted
-rwxrwxr-x    1 app      app           2756 Mar 24 07:20 exclude-tokens.txt
-rwxrwxr-x    1 app      app          70848 Mar 24 07:20 ham-samples.txt
-rwxrwxr-x    1 app      app         103724 Mar 24 07:20 spam-samples.txt
-rwxrwxr-x    1 app      app            315 Mar 24 07:20 stop-words.txt
-rw-r--r--    1 app      app          45056 Mar 24 11:09 tg-spam.db
/srv $
```

**Comment by umputun** [comment]
i think you missed the part about mounting the samples directory, pls see readme

_not a bug report but rather FAQ type of question, moved to discussions_

**Comment by kuhytitusdlz** [comment]
I don't have ham-samples or spam-samples files, and thought, if they not exists, than tg-spam should create it. I created them myself, and all works correctly now, thanks.

**Reply by umputun** [reply]
This is not how it is supposed to work. The system should create a file in this case. It was a side-effect bug introduced by https://github.com/umputun/tg-spam/pull/64. 

The fix has been pushed to the master branch and will be released as a hotfix soon.
